### PR TITLE
Remove ZSWAP from kernel configs

### LINKED ISSUE #1145: Compile zbud into the kernel used for running aarch64 tests
Missing module causes the following warnings when booting on aarch64:
```
[    2.353087] zswap: default zpool zbud not available
[    2.386900] zswap: pool creation failed
```

### PR DESCRIPTION
**Pull request by dianpopa**
## Reason for This PR

Fixes #1145.

## Description of Changes

* new working kernel configs for arm64 (kernel in S3 is now compiled with this)
* x86_64 config with balloon skipped features and zswap removed

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: resources/microvm-kernel-arm64.config [hunk 1]
```diff
@@ -193,7 +193,6 @@ CONFIG_SIGNALFD=y
 CONFIG_TIMERFD=y
```
Comment (raduiliescu) [inline resources/microvm-kernel-arm64.config:193]: There is a huge difference in the kernel configs and zswap is just a small part.  Why all these changes? How did you generate it and from where you started?

Why do we need a different config with initramfs, I guess if you don't specify initramfs at startup it shouldn't add any penalty even if the config has **support** for it.

Comment (dianpopa) [reply resources/microvm-kernel-arm64.config:193]: Maybe I should have mentioned it, apart from removing ZSWAP, I also tried to bring the arm64 config as close to the x86_64 config. I mentioned this in the commit associated. One major difference between the 2 was that the arm64 config had a lot of loadable modules I.e (=m) while x86_64 did not have any and so I tried to make the 2 as similar as possible.

Comment (dianpopa) [reply resources/microvm-kernel-arm64.config:193]: I will try to see if using initrd enabled kernel will work even if no initrd is offered.

Comment (dianpopa) [reply resources/microvm-kernel-arm64.config:200]: Indeed, we do not need a different config with initramfs, @raduiliescu take another look.

**Review by georgepisaltu — APPROVED** [review]

**Review by raduiliescu — APPROVED** [review]

File: resources/microvm-kernel-x86_64.config [hunk 2]
```diff
@@ -543,6 +541,8 @@ CONFIG_MEMORY_HOTPLUG_SPARSE=y
 CONFIG_MEMORY_HOTREMOVE=y
 CONFIG_SPLIT_PTLOCK_CPUS=4
 CONFIG_ARCH_ENABLE_SPLIT_PMD_PTLOCK=y
+CONFIG_MEMORY_BALLOON=y
+CONFIG_BALLOON_COMPACTION=y
```
Comment (sandreim) [inline resources/microvm-kernel-x86_64.config:545]: This is useful when guest is using hugepages as it helps improving memory defragmentation but at the cost of CPU overhead of an extra thread doing background work: https://elixir.bootlin.com/linux/latest/source/drivers/virtio/virtio_balloon.c#L753. 

Are we sure we want to have this by default ? I would think about a separate kernel configuration tuned for a specific usecase that requires guest huge page support.

Comment (sandreim) [reply resources/microvm-kernel-x86_64.config:545]: @georgepisaltu what is your thinking about this ?

Comment (dianpopa) [reply resources/microvm-kernel-x86_64.config:545]: I already talked to @georgepisaltu , these options are automatically selected by VIRTIO_BALLON. George removed them manually when updating the config with VIRTIO_BALLON.

**Pull request merged**